### DISCUSSION TITLE
Strange behaviour of cc65 compiler when breaking from nested loops

### DISCUSSION THREAD

**Discussion by ceptimus** [picocomputer/community#99] [Q&A] [answered]
My code is searching for an entry in a 2-dimensional grid, and I when I find the first match, I want to break out of the loops, retaining the coordinates of the match.

So I used what I thought was a fairly standard C pattern, but the results were weird.  I've reproduced the behaviour in the simplified example below:

`
#include <rp6502.h>
#include <stdio.h>

void main() {
    int i, j, found;

    found = 0; // found is a flag that, when set, should cause the loops to terminate
    for (i = 0; i < 5; i++) {
        for (j = 0; j < 5; j++) {
            printf("%d, %d\n", i, j);
            if (i == 1 && j == 1) {
                found = 1;
                break; // break out of inner loop
            }
            if (found) break; // break out of outer loop
        }
    }
    printf("Final i, j: %d, %d\n", i, j);
}
`

Resulting output is:
0, 0
0, 1
0, 2
0, 3
0, 4
1, 0
1, 1
2, 0
3, 0
4, 0
Final i, j: 5, 0

I would have expected:
0, 0
0, 1
0, 2
0, 3
0, 4
1, 0
1, 1
Final i, j: 1, 1

Is this a known bug, or am I just confused about the way I think C should work?  I fixed my code to work in a different way, but I thought I'd ask the community for their comments.

**Comment by d0plr** [comment, marked as answer]
Hi,
cc65 behaves correctly here.  Your 2d `if` statement should be moved one line down for it to work as intended.

**Reply by ceptimus** [reply]
D'oh!  How stupid of me!  Thanks for your help.
